sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala

```diff
@@ -128,6 +129,38 @@ class RapidsExecutorPlugin extends ExecutorPlugin with Logging {
         ShimLoader.setSparkShimProviderClass(conf.shimsProviderOverride.get)
       }
 
+      // Compare if the cudf version mentioned in the classpath is equal to the version which plugin
+      // expects. If there is a version mismatch, throw error. This check can be disabled by
+      // setting this config spark.rapids.cudf-version-override=true
+      if (!conf.cudfVersionOverride) {
+        val props = new Properties
+        val cudfClassLoader = classOf[ai.rapids.cudf.ColumnVector].getClassLoader
+        try {
+          props.load(cudfClassLoader.getResourceAsStream("cudf-java-version-info.properties"))
+        } catch {
+          case t: Throwable => {
+            logError("cudf properties file not found.", t)
+          }
+        }
+        val cudfVersion = props.get("version").toString
+
+        val pluginClassLoader = classOf[com.nvidia.spark.SQLPlugin].getClassLoader
+        try {
+          props.load(pluginClassLoader.getResourceAsStream("rapids4spark-version-info.properties"))
+        } catch {
+          case t: Throwable => {
+            logError("plugin properties file not found.", t)
+          }
+        }
+        val expectedCudfVersion = props.get("cudf_version").toString
+
+        if (cudfVersion != expectedCudfVersion) {
+          logError("Cudf version in the classpath is different. Found " + cudfVersion +
+            ", Plugin expects " + expectedCudfVersion)
+          System.exit(1)
```

Comment:
Why exit and not throw an exception?

Reply:
Probably for the same reason there's a `System.exit` below.  Rather than duplicate the System.exit here, I agree with @revans2 that instead of logging and exiting, an exception should be thrown with the message _instead_ of logging and let the outer `catch` handle logging it and shutting down in an appropriate manner.  That way later when the `System.exit` hack is fixed we don't have to fix it in multiple places.

Reply:
Yes, I put `System.exit` as it was below. Have modified it to throw exception with message. Please let me know if more information needs to be added in the message something like the command to set SPARK_CUDF_JAR. I did not add it here as it is mentioned in the readme.